infrastructure-playbooks/shrink-mgr.yml

```diff
@@ -0,0 +1,128 @@
+---
+# This playbook shrinks the Ceph manager from your cluster
+#
+# Use it like this:
+# ansible-playbook shrink-mgr.yml -e mgr_to_kill=ceph-mgr1
+#     Prompts for confirmation to shrink, defaults to no and
+#     doesn't shrink the cluster and yes shrinks the cluster.
+#
+# ansible-playbook -e ireallymeanit=yes|no shrink-mgr.yml
+#     Overrides the prompt using -e option. Can be used in
+#     automation scripts to avoid interactive prompt.
+
+
+- name: gather facts and check the init system
+  hosts:
+    - "{{ mon_group_name | default('mons') }}"
+    - "{{ mgr_group_name | default('mgrs') }}"
+  become: true
+  tasks:
+    - debug:
+        msg: gather facts on all Ceph hosts for following reference
+
+- name: confirm if user really meant to remove manager from the ceph cluster
+  hosts: localhost
+  become: true
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to shrink the cluster?
+      default: 'no'
+      private: no
+  pre_tasks:
+    - import_role:
+        name: ceph-defaults
+
+    - import_role:
+        name: ceph-facts
+
+    - name: exit playbook, if only one manager is present in cluster
+      fail:
+        msg: "You are about to shrink the only manager present in the cluster."
+      when: groups[mgr_group_name] | length | int < 2
```

Comment:
this is incomplete/incorrect, there's new behavior as of stable-4.0 :

mgrs can be deployed even when `[mgrs]` group is empty, it will collocate an mgr daemon on each deployed monitor.
It means you can have 3 or 5 mgrs even when `groups[mgr_group_name] | length | int < 2` is `True`

Reply:
Replaced this check by a check for number of standby MGRs the cluster has - https://github.com/ceph/ceph-ansible/pull/4150/commits/fba8a6c99d13e7bc4254d7b7c338327b6c526372#diff-d8040e37eafe0aad1ca8e16e572bfd64R52